podcasts/New Search/Views/Results/SubscribeButtonView.swift

```diff
@@ -0,0 +1,50 @@
+import SwiftUI
+import PocketCastsDataModel
+import PocketCastsServer
+
+struct SubscribeButtonView: View {
+    @State var isSubscribed = false
+
+    var podcastUuid: String
+
+    init(podcastUuid: String) {
+        self.podcastUuid = podcastUuid
+    }
+
+    var body: some View {
+        Button(action: {
+            if !isSubscribed {
+                withAnimation {
+                    isSubscribed = true
+                    subscribe()
+                }
+            }
+        }) {
+            if isSubscribed {
+                Image("discover_subscribed_dark")
+            } else {
+                Image("discover_subscribe_dark")
+            }
+        }
+        .buttonStyle(SubscribeButtonStyle())
+        .onAppear {
+            isSubscribed = DataManager.sharedManager.findPodcast(uuid: podcastUuid) != nil
+        }
+    }
+
+    private func subscribe() {
```

Comment:
I'm debating whether I should create a model or not for this.

While I think this is the correct approach, should we do something so simple? Open to feedback.